### DISCUSSION TITLE
How to debug Teal in VSCode?

### DISCUSSION THREAD

**Discussion by ThisisGame** [teal-language/tl#663] [Q&A] [answered]
Do I need to compile teal into Lua and then use the Lua debugging extension in VSCode?
Thank you for your answer.

**Comment by lenscas** [comment, marked as answer]
No idea how the vscode extension works but that sounds like one way.

Any debugger that puts the debug statements directly in the code probably works out of the box (with a .d.tl file to tell teal what is inside such a library)

**Reply by ThisisGame** [reply]
Thanks for your answer.
The debugging method of teal is indeed the same as that of lua, and any lua debugger can debug teal.

**Reply by ThisisGame** [reply]
I successfully debugged in ZeroBrain.

https://github.com/ThisisGame/cpp-game-engine-book/blob/main/imgs/teal/debug_in_zerobrane.gif

**Comment by Aire-One** [comment]
I am able to debug Teal code from VSCode with this extension: https://github.com/tomblind/local-lua-debugger-vscode

You need to add the `"scriptFiles": ["${workspaceFolder}/**/*.tl"]` settings in your `launch.json` config file, and the VSCode settings `"debug.allowBreakpointsEverywhere": true` to allow breakpoints to be respectively "listened to" and placed in Teal files (+ follow instructions in the README).

**Reply by ThisisGame** [reply]
Thanks for your answer.
The debugging method of teal is indeed the same as that of lua, and any lua debugger can debug teal.
